server/data.py

```diff
@@ -139,53 +139,27 @@ def _get_het_hom_index(self, data, index, use_gt, gp_thres, calc_info):
                 gt = s[0]
                 if gt == '1|1' or gt == '1/1':
                     hom_alt_i.append(i)
-                elif not (gt == '0|0' or gt == '0/0'):
+                elif (gt == '1|0' or gt == '1/0' or gt == '0|1' or gt == '0/1'):
                     het_i.append(i)
+                elif (gt == '0|0' or gt =='0/0'):
+                    wt_hom_i.append(i)
+                else:
+                    # if there are missing calls in the raw chip data
+                    missing_i.append(i)
             else:
-                if gp[2] >= gp_thres:
+                if gp[2] >= float(gp_thres):
```

Comment:
this does not go correctly. you first have to take the most probable genotype (i.e GT) and check if the GP of that genotype is above the threshold and then count it as that genotype. Now if user does not want to do gp_threshold filtering, all genotypes would be hom_alt